syncutil/detach.go

```diff
@@ -0,0 +1,38 @@
+package syncutil
```

Comment:
You reckon? A detached context would normally only be used in asynchronous work as you're looking to background some work that will live longer than the request that fires it off. So it existing in a sync package makes sense to me.

Reply:
> The naming scheme is [type]util or [pkgname]util. If there already is a *util packge in stdlib it's named utilx (e.g. ioutilx).

If we're being sticklers, we're working with `context` package here rather than `sync` 😅 

And maybe I'm raving here, but arguably what detached context achieves is the opposite of what `sync` package does - rather than ensuring multiple asynchronous jobs can share a resource in a synchronous manner, you're spawning a copy of a synchronous resource to be used in an asynchronous manner. 🤔

Reply:
That makes sense. I'll move this to a contextutil package :+1: